transformer_engine/pytorch/module/grouped_linear.py

```diff
@@ -153,12 +170,14 @@ def forward(
 
             _ = fp8_grouped_gemm(
                 [w._data for w in weights_fp8],
-                fp8_meta["scaling_fwd"].scale_inv,
-                _GEMM_WEIGHT,
```

Comment:
Hi @timmoon10, I want to confirm this is a bug, right? FP8 weights are casted in `get_fp8_workspace` in which a standalone `scale_inv` is set for each Float8Tensor instead of using `fp8_meta["scaling_fwd"].scale_inv`.

I think it's a bit confusing that in `get_fp8_workspace`:
- `with_transpose`: Only `Float8Tensor._scale_inv` is updated.
- not `with_transpose`: `fp8_meta.scale_inv[fp8_meta_index]` is updated and then copied to `Float8Tensor._scale_inv`.
https://github.com/NVIDIA/TransformerEngine/blob/47caafb28fd3f8c99ada92f9d7504715d268aa91/transformer_engine/pytorch/module/base.py#L997

Reply:
What is the bug you are noticing? You are changing the API for `fp8_grouped_gemm`, but other than that it looks like using `fp8_meta` would be correct (but not recommended).

Reply:
When casting weights to FP8 in `get_fp8_workspace()`, the `scale_inv`s of weight tensors are written to the private ones in Float8Tensors, while the `scale_inv`s in `fp8_meta` are updated after the first forward step is done. So for the first micro batch, they don't match.